framework/src/modules/interoperability/base_interoperability_store.ts

```diff
@@ -181,14 +183,89 @@ export abstract class BaseInteroperabilityStore {
 		await terminatedOutboxSubstore.setWithSchema(chainID, terminatedOutbox, terminatedOutboxSchema);
 	}
 
+	public async createTerminatedStateAccount(chainID: number, stateRoot?: Buffer): Promise<boolean> {
+		const chainIDAsStoreKey = getIDAsKeyForStore(chainID);
+		let terminatedState: TerminatedStateAccount;
+		const chainSubstore = this.getStore(MODULE_ID_INTEROPERABILITY, STORE_PREFIX_CHAIN_DATA);
+		const isExist = await this.chainAccountExist(chainIDAsStoreKey);
+
+		if (stateRoot) {
+			if (isExist) {
```

Comment:
you can combine the condition instead of nesting,
`if (stateRoot && isExist)`

Reply:
To do that that we'd also need to add one more condition `if (stateRoot && !isExist)` (for setting `termiatedState`) in addition to the two conditions you mentioned. I'm not sure if it'd improve the efficiency.

Reply:
we try to avoid nesting of conditions as much as possible but if it doesn't improve much then we can keep as it is